Client/Application/Delegates/AppDelegate.swift

```diff
@@ -258,9 +258,13 @@ class AppDelegate: UIResponder, UIApplicationDelegate, UIViewControllerRestorati
         if isFirstLaunch {
             profile?.searchEngines.regionalSearchEngineSetup()
         }
+        
         if let urp = UserReferralProgram.shared {
             if isFirstLaunch {
-                urp.referralLookup { referralCode, offerUrl in
+                let refCode = UserReferralProgram.sanitize(input: UIPasteboard.general.string)
+                if refCode != nil { UIPasteboard.general.clearPasteboard() }
+                
+                urp.referralLookup(refCode: refCode) { referralCode, offerUrl in
```

Comment:
If this call fails for some reason (spotty internet, loss of connection, server went down, etc.), I assume we don't have any retry logic? Given that we clear out the pasteboard and don't save the parsed ref code

Reply:
We actually do save it before the network request here: https://github.com/brave/brave-ios/pull/2535/files#diff-60f6d7911cf19ceb7bcfcde55f9bc740R87
And the network request should be re-attempted, if onboarding not completed, and we pull from prefs to use the actual referral code. I will add a comment there to help clarify.